### DISCUSSION TITLE
Question about GSEA in the eGRN compilation part

### DISCUSSION THREAD

**Discussion by tianchen2019** [aertslab/scenicplus#65] [Q&A]
I have some questions about the eGRN compilation part.
 Is GSEA calculated for each TF-region-gene module? Which genes are in the ranking list L?
My understanding is that the ranking list is the importance score of all downstream genes corresponding to a transcription factor, and set is the downstream genes in the module corresponding to this TF. I don't know if my understanding is correct.
 I am not very clear about the detailed process of this part. I hope you can help me out.
 Thanks

**Comment by SeppeDeWinter** [comment]
Hi @tianchen2019 

I transferred your question to the discussions tab because it's not really an issue with the code but a question.

For an explanation of this step I would like to refer you to the API page in the read the docs: https://scenicplus.readthedocs.io/en/latest/api.html#gsea-based-approach.

In a nutshell:
The ranking is based on the TF-to-gene importance scores and the gene set is obtained by binarising region-to-gene links for all regions which are determined to have a motif for the TF.

For a more detailed (and illustrated) explanation, please refer to the API link above.

I hope this clarifies things. If not, feel free to drop a message below.

Best,

Seppe
